### ISSUE TITLE
Invalid SQL generated by SQLite provider for GroupBy().Where() for decimals 

### ISSUE THREAD

**Issue by julielerman** [dotnet/efcore#12715]
_(Sorry if this is a dupe. I did try to find a related issue)_
I do understand that EFCore maps decimals to text fields and based on https://github.com/aspnet/EntityFrameworkCore/issues/7232 that evaluation may not always go well.

However I think 
When a groupby query filters on an aggregate of a decimal, the SQL represents the value to be compared as a string.

unit.Cost is a decimal property. In the database it's TEXT.

 ```
var unitGrouping = context.Units
                    .GroupBy(u=>u.BrewerTypeId)
                     .Where(g => g.Sum(u => u.Cost) > 20)
                    .Select(g => new
                    {
                        BrewerTypeId= g.Key,
                        Cost = g.Sum(u => u.Cost),
                        Count = g.Count()
                    }).ToList();
```
The SQL:

```
 SELECT "u"."BrewerTypeId", SUM("u"."Cost") AS "Cost", COUNT(*) AS "Count"
      FROM "Units" AS "u"
      GROUP BY "u"."BrewerTypeId"
      HAVING SUM("u"."Cost") > '20.0'

```
In this format, no data is being returned from my database because SUM is not returning text.

 If 20 came in without the quotes, e.g.

```
SELECT "u"."BrewerTypeId", SUM("u"."Cost") AS "Cost", COUNT(*) AS "Count"
      FROM "Units" AS "u"
      GROUP BY "u"."BrewerTypeId"
      HAVING SUM("u"."Cost") > 20.0
```

Then SQLite is happy  and returns the correct results.

Can this be achieved when composing the SQL?

### Further technical details
EF Core version: 2.1.1
Database Provider: Microsoft.EntityFrameworkCore.Sqlite
Operating system: Windows 10 latest
IDE:  Visual Studio 2017 15.7.5
Note: I'm using SQLIte/SQL CE toolbox and it's query editor to test the raw SQL generated by EF Core.

**Comment by smitpatel** [comment]
Probably need to do something with type inference since SQLite type inference is somewhat different.

**Comment by julielerman** [comment]
Who? Me when I build my query or something internal in the provider?

**Comment by bricelam** [comment]
I believe I made this client eval on 2.2. There's another issue somewhere tracking the ability to generate a CAST to double from decimal...

**Comment by smitpatel** [comment]
@bricelam made it client eval and it fails in group by translation.

Blocked on #12089

**Comment by divega** [comment]
Clearing milestone to discuss propose-close in regular triage.

**Comment by ajcvickers** [comment]
Note from triage: the way to make this work in the new query pipeline is to cast to a double in the query:
```C#
Cost = g.Sum(u => (double)u.Cost),
```
The plan is for this to generate a conversion in the query such that the comparison will be correct.
